packages/contractkit/src/wrappers/Attestations.ts

```diff
@@ -1,18 +1,139 @@
+import { CeloToken } from './TokenWrapper'
+const parseSignature = SignatureUtils.parseSignature
 
 export interface AttestationStat {
   completed: number
   total: number
 }
 
+export interface IssuerAttestationState {
+  [issuer: string]: {}
+}
+
+export enum AttestationState {
+  None,
+  Incomplete,
+  Complete,
+}
+
+export interface ActionableAttestation {
+  issuer: string
+  attestationState: AttestationState
+  time: number
+  publicKey: string
+}
+
+const parseAttestationInfo = (rawState: { 0: string; 1: string }) => ({
+  attestationState: parseInt(rawState[0], 10),
+  time: parseInt(rawState[1], 10),
+})
+
+function attestationMessageToSign(phoneHash: string, account: string) {
+  const messageHash: string = Web3Utils.soliditySha3(
+    { type: 'bytes32', value: phoneHash },
+    { type: 'address', value: account }
+  )
+  return messageHash
+}
+
 export class AttestationsWrapper extends BaseWrapper<Attestations> {
+  attestationExpirySeconds = proxyCall(
+    this.contract.methods.attestationExpirySeconds,
+    undefined,
+    toNumber
+  )
+
+  getAttestationStat = proxyCall(
+    this.contract.methods.getAttestationStats,
+    tupleParser(PhoneNumberUtils.getPhoneHash, (x: string) => x),
+    (stat) => ({ completed: toNumber(stat[0]), total: toNumber(stat[1]) })
+  )
+
+  getWalletAddress = proxyCall(this.contract.methods.getWalletAddress)
+
   setAccountDataEncryptionKey = proxySend(
     this.kit,
     this.contract.methods.setAccountDataEncryptionKey
   )
 
+  setWalletAddress = proxySend(this.kit, this.contract.methods.setWalletAddress)
+
+  async approveAttestationFee(token: CeloToken, attestationsRequested: number) {
+    const fee = await this.attestationFeeRequired(token.address, attestationsRequested)
+    return token.approve(this.address, fee.toString())
+  }
+
+  async attestationFeeRequired(tokenAddress: string, attestationsRequested: number) {
+    const attestationFee = await this.contract.methods.getAttestationRequestFee(tokenAddress).call()
+    return new BigNumber(attestationFee).times(attestationsRequested)
+  }
+
+  async getActionableAttestations(
+    phoneNumber: string,
+    account: string
+  ): Promise<ActionableAttestation[]> {
+    const phoneHash = PhoneNumberUtils.getPhoneHash(phoneNumber)
+    const expirySeconds = await this.attestationExpirySeconds()
+    const nowInUnixSeconds = Math.floor(new Date().getTime() / 1000)
+
+    const issuers = await this.contract.methods.getAttestationIssuers(phoneHash, account).call()
+    const issuerState = Promise.all(
+      issuers.map((issuer: any) =>
+        this.contract.methods
+          .getAttestationState(phoneHash, account, issuer)
+          .call()
+          .then(parseAttestationInfo)
+      )
+    )
+    const publicKeys = Promise.all(
+      issuers.map((issuer) => this.contract.methods.getDataEncryptionKey(issuer).call())
+    )
+
+    return ((zip(issuers, await issuerState, await publicKeys) as unknown) as Array<
```

Comment:
what the problem with the typing?

Reply:
zip's proper type is that any of them could be unknown if the length of the arrays are not equal